### ISSUE TITLE
backupccl: remove cloud I/O from job planning transaction

### ISSUE THREAD

**Issue by shermanCRL** [cockroachdb/cockroach#66502]
@miretskiy has pointed out that we perform I/O to cloud storage while holding open a transaction during planning.  In particular, this can lock the Jobs table.

Of course, I/O can take an arbitrary amount of time, and is not ’transactional’. Overall, seems like a bad pattern, and a recipe for delays, lockups, or failures.

We are not currently aware of the impact, but we suspect some bad behavior may be attributable to this.

Epic CRDB-7075

Jira issue: CRDB-8068

**Comment by miretskiy** [comment]
I think historical reasons for doing any of the external IO are no longer applicable.
In particular, the only reason I can think of to access external IO is so that we can return an error to the end user earlier, without creating job record.
I think the above rationale is no longer applicable:
  1. Creation of job records is not THAT onerous.
  2. We have `DETACHED` option now (a reasonably recent behavior) where returning an error when resumer starts vs during planning is not that different.
  3. Jobs infra refactoring made results reporting much simpler.  In particular, `JobResultsReporter` is perfectly capable of reporting any errors which we report now.
In short, I think nothing outside of writing a minimal job record should be done in the planning phase.
